src/radical/pilot/utils/misc.py

```diff
@@ -219,5 +219,108 @@ def get_resource_job_url(resource: str,
     return ru.Url(rcfg.schemas[schema]['job_manager_endpoint'])
 
 
+# ------------------------------------------------------------------------------
+#
+def convert_slots_to_new(slots, log=None):
+
+    from ..resource_config import Slot, RO
+
+    if not slots:
+        return slots
+
+    new_slots = list()
+    for slot in slots:
+
+        # check slot type: only new slots have a version field
+        if slot.get('version'):
+            new_slots.append(slot)
+            continue
+
+        cores = slot['cores']
+        if cores:
+            if isinstance(cores[0], int):
+                cores = [RO(index=i, occupation=1.0)
+                         for i in slot['cores']]
+            elif isinstance(cores[0], dict):
+                cores = list()
+                for ro in slot['cores']:
+                    i = ro['index']
+                    o = ro['occupation']
+                    cores.append(RO(index=i, occupation=o))
+            else:
+                cores = [RO(index=i, occupation=o)
+                         for i,o in slot['cores']]
+
+
+        gpus = slot['gpus']
+        if gpus:
+            if isinstance(gpus[0], int):
+                gpus  = [RO(index=i, occupation=1.0)
+                         for i in slot['gpus']]
+            elif isinstance(gpus[0], dict):
+                gpus = list()
+                for ro in slot['gpus']:
+                    i = ro['index']
+                    o = ro['occupation']
+                    gpus.append(RO(index=i, occupation=o))
+            else:
+                gpus  = [RO(index=i, occupation=o)
+                         for i,o in slot['gpus']]
+
+        new_slot = Slot(cores=cores,
+                        gpus=gpus,
+                        lfs=slot['lfs'],
+                        mem=slot['mem'],
+                        node_index=slot['node_index'],
+                        node_name=slot['node_name'])
+        new_slots.append(new_slot)
+
+    return new_slots
+
+
+# ------------------------------------------------------------------------------
+#
+def convert_slots_to_old(slots, log=None):
+
+    if not slots:
+        return slots
+
+    old_slots = list()
+    for slot in slots:
+
+        # if slots do not have a version field, they are old
+        if not slot.get('version'):
+            old_slots.append(slot)
+            continue
+
+        cores = slot['cores']
+        if cores:
+            if isinstance(cores[0], int):
+                cores = [[c] for c in cores]
+            elif isinstance(cores[0], dict):
+                cores = [[ro['index']] for ro in cores]
+            else:
+                cores = [[ro.index] for ro in cores]
```

Comment:
FYI, if `cores[0]` is `RO` object, then `isinstance(cores[0], dict)` will return `True`
(`RO` is based on `TypedDict` is based on `dict`)

Reply:
Thanks - explicitly checking for `RO` now.